autorag/nodes/retrieval/bm25.py

```diff
@@ -126,12 +148,15 @@ def __init__(self, project_dir: str, *args, **kwargs):
 		if bm25_tokenizer is None:
 			bm25_tokenizer = "porter_stemmer"
 		bm25_path = os.path.join(self.resources_dir, get_bm25_pkl_name(bm25_tokenizer))
-		assert (
-			bm25_path is not None
-		), "bm25_path must be specified for using bm25 retrieval."
-		assert os.path.exists(
-			bm25_path
-		), f"bm25_path {bm25_path} does not exist. Please ingest first."
+		
+		if not os.path.exists(bm25_path):
+			print(f"bm25_path {bm25_path} does not exist. Ingesting corpus...")
+            # Assuming `corpus_data` is provided in kwargs
+			corpus_data = pd.read_parquet("data/corpus.parquet")
+			if corpus_data is not None:
+				bm25_ingest(bm25_path, corpus_data, bm25_tokenizer)
+			else:
+				raise ValueError("Corpus data must be provided to ingest and create the BM25 corpus.")
```

Comment:
I think try to re-ingest bm25 is here looks not great.
Because someone just can make a mistake to type a BM25 corpus path, but if it re-ingest whole corpus, its computation will be wasted + It is unexpected result.